pkg/cmd/init/run.go

```diff
@@ -23,33 +23,52 @@ import (
 	"github.com/okteto/okteto/pkg/k8s/pods"
 	"github.com/okteto/okteto/pkg/k8s/replicasets"
 	"github.com/okteto/okteto/pkg/k8s/services"
+	"github.com/okteto/okteto/pkg/linguist"
+	"github.com/okteto/okteto/pkg/log"
 	"github.com/okteto/okteto/pkg/model"
 	"github.com/okteto/okteto/pkg/okteto"
 	appsv1 "k8s.io/api/apps/v1"
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
 var (
 	componentLabels []string = []string{"app.kubernetes.io/component", "component", "app"}
 )
 
 // SetDevDefaultsFromDeployment sets dev defaults from a running deployment
-func SetDevDefaultsFromDeployment(ctx context.Context, dev *model.Dev, d *appsv1.Deployment, container string) error {
-	c, _, err := k8Client.GetLocalWithContext(dev.Context)
+func SetDevDefaultsFromDeployment(ctx context.Context, dev *model.Dev, d *appsv1.Deployment, container, language string) error {
+	c, config, err := k8Client.GetLocalWithContext(dev.Context)
 	if err != nil {
 		return err
 	}
 
-	setAnnotationsFromDeployment(dev, d)
-	setNameAndLabelsFromDeployment(ctx, dev, d)
-
 	pod, err := getRunningPod(ctx, d, container, c)
 	if err != nil {
 		return err
 	}
 
+	language = linguist.NormalizeLanguage(language)
+	updateImageFromPod := false
+	switch language {
+	case linguist.Javascript:
+		updateImageFromPod = pods.HasNodeModules(ctx, pod, container, config, c)
```

Comment:
for node we should also do the `init from volume` thing you implemented the other day, so they don't get replaced.

Reply:
it is already the default behavior